src/main.ts

```diff
@@ -40,7 +40,7 @@ const showIngredients = () => {
 
 const selectIngredient = (event: JQuery.ClickEvent) => {
   if (selectedIngredients.includes(parseInt($(event.target).attr('id')))) {
-    selectedIngredients = selectedIngredients.filter(ing => !ing.includes(parseInt($(event.target).attr('id'))))
+    selectedIngredients = selectedIngredients.filter(ing => ing !== (parseInt($(event.target).attr('id'))))
```

Comment:
you are computing it on every iteration

Reply:
this changed list of selectedIngredients each time some ingredient is unchecked